edit.js

```diff
@@ -12,10 +14,73 @@ appliesToEverythingTemplate.innerHTML = t("appliesToEverything") + ' <button cla
 var sectionTemplate = document.createElement("div");
 sectionTemplate.innerHTML = '<label>' + t('sectionCode') + '</label><textarea class="code"></textarea><br><div class="applies-to"><label>' + t("appliesLabel") + ' <img class="applies-to-help" src="help.png" alt="' + t('helpAlt') + '"></label><ul class="applies-to-list"></ul></div><button class="remove-section">' + t('sectionRemove') + '</button><button class="add-section">' + t('sectionAdd') + '</button>';
 
+document.addEventListener("change", function(event) {
+	var node = event.target;
+	if (node.type && !node.form) { // INPUTs that aren't in a FORM are stylesheet
+		switch (node.type) {
+			case "checkbox":
+				setCleanItem(node, node.checked === node.defaultChecked);
+				break;
+			case "text":
+			case "select-one":
+			case "select-multiple":
+				setCleanItem(node, node.value === node.defaultValue);
+				break;
+		}
+	}
+});
+
+// Set .dirty on stylesheet contributors that have changed
+var items = {};
+function isCleanItem(node) {
+	return items[node.id];
+}
+function setCleanItem(node, clean) {
+	var id = node.id;
+	if (!id) id = node.id = Date.now().toString(32).substr(-6);
+	items[id] = clean;
+
+	if (clean) node.classList.remove("dirty");
+	else node.classList.add("dirty");
```

Comment:
Until recently there were only 8 braceless ifs in Stylish ([2 in install.js](https://github.com/JasonBarnabe/stylish-chrome/blob/3fddfae15a7e34d2f8c32c0d58a5e644a4dc97e4/install.js#L40-L43) and [6 in manage.js](https://github.com/JasonBarnabe/stylish-chrome/blob/b019038f5701b67d475a36275fcbe69ac5d6ee04/manage.js#L59-L72)), then @hideheader added another 8 or more if the `else` parts are also counted ([2 in edit.js](https://github.com/JasonBarnabe/stylish-chrome/blob/284d4b8ec89babc088b2b8b6eed839ec297b0ec9/edit.js#L80-L83), [1 in popup.js](https://github.com/JasonBarnabe/stylish-chrome/blob/f8600c2fb3b377dc185d07533be0b80da4865a0b/popup.js#L57), [4 in storage.js](https://github.com/JasonBarnabe/stylish-chrome/blob/e7f6c62e0178905233c0c420cc8de758a22fd3af/storage.js#L196-L215), [1 in manage.js](https://github.com/JasonBarnabe/stylish-chrome/blob/a0482d3f0b903adefbc2e092741d6ea4d5dcade0/manage.js#L357)).

For personal _write-only_ small stuff I use the compact one-line braceless form. Some large projects also use it (CodeMirror for example). However generally it's considered bad practice because it may lead to errors when another statement is added to the braceless block without being cautious enough to add the braces.

So, @hideheader, @JasonBarnabe, should we enforce braces on `if-else` or not? Should all one-line `if() action;` be expanded to 3 lines?